callbacks/control_bar.py

```diff
@@ -800,3 +801,47 @@ def open_controls_drawer(n_clicks, is_opened):
             return no_update, {"display": "none"}
         return no_update, {}
     return no_update, no_update
+
+
+@callback(
+    Output("result-selector", "data"),
+    Output("result-selector", "value"),
+    Output("result-selector", "disabled"),
+    Output("show-result-overlay-toggle", "checked"),
+    Output("show-result-overlay-toggle", "disabled"),
+    Output("seg-result-opacity-slider", "disabled"),
+    Input("project-name-src", "value"),
+    Input("show-result-overlay-toggle", "checked"),
+    State("result-selector", "disabled"),
+    State("seg-result-opacity-slider", "disabled"),
+)
+def populate_classification_results(
+    image_src, toggle, dropdown_enabled, slider_enabled
+):
+    results = []
+    value = None
+    checked = False
+    disabled_dropdown = True
+    disabled_toggle = True
+    disabled_slider = True
+    if ctx.triggered_id == "show-result-overlay-toggle":
+        results = no_update
+        value = no_update
+        checked = no_update
+        disabled_dropdown = dropdown_enabled
+        disabled_toggle = False
+        disabled_slider = slider_enabled
+    else:
+        results = [
+            item
+            for item in get_data_project_names()
+            if ("seg" in item and image_src in item)
+        ]
+        if results:
```

Comment:
For some reason, this does not yield a populated dropdown when I am testing against the Tiled instance on Spin. 

However, I am interestingly now also hitting the timeout problems that you have been seeing before.

Those don't show up in a fully reproducible way, but I seem to hit them most consistently when I switch projects. 

I wonder if the timeout for retrieving data, also affects this callback, then causing for the segmentation result not to show up correctly.

Reply:
An update on this: The timeout was indeed the issue. With no time-outs the dropdown gets correctly populated and overlaying works great too.